File: app/src/main/java/com/tosh/poolandroid/RoomDb/UserDao.java
package com.tosh.poolandroid.RoomDb;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.tosh.poolandroid.Model.User;

import java.util.List;

@Dao
public interface UserDao {
    @Insert
    void insert(UserRoom user);

    @Delete
    void delete(UserRoom user);

    @Query("SELECT * FROM user_table ORDER BY email")
    LiveData<List<UserRoom>>  getUserDetails();


}

File: app/src/main/java/com/tosh/poolandroid/model/database/UserDao.kt
package com.tosh.poolandroid.model.database

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface UserDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertDetails(userEntity: UserEntity)

    @Query("SELECT * FROM user_table")
    fun getUserDetails(): LiveData<List<UserEntity>>

    @Query("DELETE FROM user_table")
    suspend fun deleteUser()


}